### PR TITLE
WIP vicadmin authentication

### PR DESCRIPTION
**Pull request by gigawhitlocks**
Working: authentication w/ client-side certificates e.g. `wget --certificate=ian-vch/cert.pem --private-key=ian-vch/key.pem --no-check-certificate https://192.168.218.212:2378/logs/vicadmin.log`

Stubbed out: username/password authentication. Plan to implement this part tomorrow. In the current version authentication is impossible w/o a certificate & the user will be redirected to `/authentication`

Also: small refactor to get all routes redirected through `s.HandleFunc` so that we can easily apply authentication -- might make some changes w/r/t how this is designed depending on how everything shakes out tomorrow while I'm doing username/password auth, but the idea is to make it obvious and easy to specify that a user must be authenticated to view a resource if he must, and likewise if the resource does not require authentication.

@sgairo I seem to recall you asking to be tagged once I opened a PR

### CODE REVIEW

File: cmd/vicadmin/server.go [hunk 1]
```diff
@@ -274,6 +282,19 @@ func (s *server) tailFiles(res http.ResponseWriter, req *http.Request, names []s
 	}
 }
 
+func (s *server) authPage(res http.ResponseWriter, req *http.Request) {
```
Comment (gigawhitlocks) [inline cmd/vicadmin/server.go]: this is just a stub for a user login page

**Comment by sgairo** [comment]
@gigawhitlocks let me know if you need anything for the auth.html (we used Clarity for dashboard)

File: cmd/vicadmin/server.go [hunk 2]
```diff
@@ -101,61 +108,62 @@ func (s *server) listenPort() int {
...
 }
 
 func (s *server) serve() error {
 	defer trace.End(trace.Begin(""))
 
 	s.mux = http.NewServeMux()
 
+	s.mux.HandleFunc("/authentication", s.authPage)
+	// NOTE:
+	// DO NOT USE s.mux.HandleFunc directly outside of HandleFunc; you will bypass authentication!
```
Comment (vburenin) [inline cmd/vicadmin/server.go]: this URL is repeated. It definitely wants to be a constant.

File: cmd/vicadmin/server.go [hunk 3]
```diff
@@ -28,12 +28,12 @@ import (
 	log "github.com/Sirupsen/logrus"
 	"github.com/docker/docker/pkg/tlsconfig"
 
+	"crypto/x509"
```
Comment (vburenin) [inline cmd/vicadmin/server.go]: this is a standard package, please move it to the top.

**Comment by hickeng** [comment]
@gigawhitlocks FYI #2629 now generates a `pfx` certificate if openssl is available, which can easily be imported into client browser.

**Comment by gigawhitlocks** [comment]
Closing this because I made a new branch w/ just the changes for client cert auth

**Pull request closed without merging**